Core/Src/main.c

```diff
@@ -27,7 +27,7 @@
 
 /* Private includes ----------------------------------------------------------*/

```

Comment:
I think working with modules gives us more flexibility, and as you can see at the bottom of the page, main.c is just a link to cpp_main(), which is our main function with C++ support. 

In the Lib folder you can find all the code for the application, I would suggest creating 2 files (.c and .h) just for this sensor. For example (bme280.h and bme280.c)

Reply:
@samkno1, I am going to create what Nico is suggesting and push the branch. This way you'll see a clear example of how modules are created. Once architectural problem is solved we can continue the discussion of a more coding style type comments.

Reply:
> > https://github.com/Niuslar/Multichannel-Temperature-Control/tree/feature/srm_bme280_refactor
> 
> @samkno1, I am going to try and refactor your code the way I see fit. If you'd like to follow, feel free. However, I recommend that you have a go yourself. If you want we can have a pair programming session sometime next week. I am back from holiday on 1st of August.

Yes, I have already begun restructuring the code. Yes, I'd appreciate that and I am available 6 PM, August 5th (Atlantic Standard Time).